src/Command/ListTasksCommand.php

```diff
@@ -102,20 +105,39 @@ protected function execute(InputInterface $input, OutputInterface $output): int
         }
 
         $symfonyStyle->success(sprintf('%d task%s found', count($tasks), count($tasks) > 1 ? 's' : ''));
-
         $table = new Table($output);
-        $table->setHeaders(['Name', 'Description', 'Expression', 'Last execution date', 'Next execution date', 'Last execution duration', 'Last execution memory usage', 'State', 'Tags']);
-        $table->addRows(array_map(fn (TaskInterface $task): array => [
-            $task->getName(),
-            $task->getDescription() ?? 'No description set',
-            $task->getExpression(),
-            null !== $task->getLastExecution() ? $task->getLastExecution()->format(DATE_ATOM) : 'Not executed',
-            (new CronExpression($task->getExpression()))->getNextRunDate()->format(DATE_ATOM),
-            null !== $task->getExecutionComputationTime() ? Helper::formatTime($task->getExecutionComputationTime() / 1000) : 'Not tracked',
-            null !== $task->getExecutionMemoryUsage() ? Helper::formatMemory($task->getExecutionMemoryUsage()) : 'Not tracked',
-            $task->getState(),
-            implode(', ', $task->getTags()) ?? 'No tags set',
-        ], $tasks));
+        $table->setHeaders(['Type', 'Name', 'Description', 'Expression',  'Last execution date', 'Next execution date', 'Last execution duration', 'Last execution memory usage', 'State', 'Tags']);
+
+        array_walk($tasks, function (TaskInterface $task) use ($table): void {
```

Comment:
If I'm right, we can use:

```php
$tasks->walk(function (TaskInterface $task) use ($table): void {
    // 
});
```

This is probably a better idea as we can iterate over the list pointer and not create a new one here 🙂

Reply:
you're right, I completely forgot about this new function: D